db/channel_cache.go

```diff
@@ -187,6 +192,22 @@ func (c *channelCache) _pruneCache() {
 		c.logs = c.logs[pruned:]
 	}
 
+	return pruned
+}
+
+func (c *channelCache) pruneCacheAge() {
+	c.lock.Lock()
+	pruned := c._pruneCacheAge()
+	c.lock.Unlock()
+	base.Debugf(base.KeyCache, "Pruned %d old entries from channel %q", pruned, base.UD(c.channelName))
+}
+
+func (c *channelCache) _pruneCacheAge() (pruned int) {
+	// time-based cache pruning doesn't make sense when MinLength == MaxLength
+	if c.options.ChannelCacheMinLength == c.options.ChannelCacheMaxLength {
```

Comment:
Can probably make this ChannelCacheMinLength >= ChannelCacheMaxLength to be extra defensive.